File: Entity/Promotion.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Promotion
 *
 * @ORM\Table(name="promotion", indexes={@ORM\Index(name="idUser", columns={"idUser"})})
 * @ORM\Entity
 */
class Promotion
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="critaire", type="text", length=65535, nullable=false)
     */
    private $critaire;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateD", type="date", nullable=false)
     */
    private $dated;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateF", type="date", nullable=false)
     */
    private $datef;

    /**
     * @var int
     *
     * @ORM\Column(name="idUser", type="integer", nullable=false)
     */
    private $iduser;

    /**
     * @var string
     *
     * @ORM\Column(name="castPhoto", type="string", length=255, nullable=false)
     */
    private $castphoto;

    /**
     * @var string
     *
     * @ORM\Column(name="reduction", type="string", length=255, nullable=false)
     */
    private $reduction;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCritaire(): ?string
    {
        return $this->critaire;
    }

    public function setCritaire(string $critaire): self
    {
        $this->critaire = $critaire;

        return $this;
    }

    public function getDated(): ?\DateTimeInterface
    {
        return $this->dated;
    }

    public function setDated(\DateTimeInterface $dated): self
    {
        $this->dated = $dated;

        return $this;
    }

    public function getDatef(): ?\DateTimeInterface
    {
        return $this->datef;
    }

    public function setDatef(\DateTimeInterface $datef): self
    {
        $this->datef = $datef;

        return $this;
    }

    public function getIduser(): ?int
    {
        return $this->iduser;
    }

    public function setIduser(int $iduser): self
    {
        $this->iduser = $iduser;

        return $this;
    }

    public function getCastphoto(): ?string
    {
        return $this->castphoto;
    }

    public function setCastphoto(string $castphoto): self
    {
        $this->castphoto = $castphoto;

        return $this;
    }

    public function getReduction(): ?string
    {
        return $this->reduction;
    }

    public function setReduction(string $reduction): self
    {
        $this->reduction = $reduction;

        return $this;
    }


}

File: src/tn/esprit/TuniShow/entity/Promotion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tn.esprit.TuniShow.entity;




public class Promotion {
    
    int id,idUser;
    String critaire,castPhoto,reduction;
    String dateD , dateF;

    public Promotion() {
    }

    public Promotion(int id, int idUser, String critaire, String castPhoto, String reduction, String dateD, String dateF) {
        this.id = id;
        this.idUser = idUser;
        this.critaire = critaire;
        this.castPhoto = castPhoto;
        this.reduction = reduction;
        this.dateD = dateD;
        this.dateF = dateF;
    }
  

    public Promotion(int idUser, String critaire, String castPhoto, String reduction, String dateD, String dateF) {
        this.idUser = idUser;
        this.critaire = critaire;
        this.castPhoto = castPhoto;
        this.reduction = reduction;
        this.dateD = dateD;
        this.dateF = dateF;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    public String getCritaire() {
        return critaire;
    }

    public void setCritaire(String critaire) {
        this.critaire = critaire;
    }

    public String getCastPhoto() {
        return castPhoto;
    }

    public void setCastPhoto(String castPhoto) {
        this.castPhoto = castPhoto;
    }

    public String getReduction() {
        return reduction;
    }

    public void setReduction(String reduction) {
        this.reduction = reduction;
    }

    public String getDateD() {
        return dateD;
    }

    public void setDateD(String dateD) {
        this.dateD = dateD;
    }

    public String getDateF() {
        return dateF;
    }

    public void setDateF(String dateF) {
        this.dateF = dateF;
    }

  
    
    
    
    
}
